pkg/storage/etcd3/watcher_test.go

```diff
@@ -227,14 +265,25 @@ func (c *testCodec) Decode(data []byte, defaults *unversioned.GroupVersionKind,
 	return nil, nil, errors.New("Expected decoding failure")
 }
 
+func testCheckEventType(t *testing.T, expectEventType watch.EventType, w watch.Interface) {
```

Comment:
@xiang90 
Your suggestion sounds actually a good idea.
I have split another function to check event type specifically.